emoji/core.py

```diff
@@ -45,3 +45,56 @@ def replace(match):
             return unicode_codes.EMOJI_UNICODE.get(match.group(1), match.group(1))
 
     return pattern.sub(replace, string)
+
+def demojize(string):
+
+    """Replace unicode emoji in a string with emoji shortcodes. Useful for storage.
+
+    :param string: String contains unicode characters. MUST BE UNICODE.
+
+        >>> import emoji
+        >>> print(emoji.emojize("Python is fun :thumbs_up_sign:"))
+        Python is fun 👍
+        >>> print(emoji.demojize(u"Python is fun 👍"))
+        Python is fun :thumbs_up_sign:
+        >>> print(emoji.demojize("Unicode is tricky 😯".decode('utf-8')))
+        Unicode is tricky :hushed_face:
+    """
+
+    # via Martijn Pieters, http://stackoverflow.com/questions/26568722/remove-unicode-emoji-using-re-in-python
+    if PY2:
+        try:
+            # Wide UCS-4 build
+            pattern = re.compile(u'['
+                u'\U0001F300-\U0001F64F'
+                u'\U0001F680-\U0001F6FF'
+                u'\u2600-\u26FF\u2700-\u27BF]+', 
+                re.UNICODE)
+        except re.error:
+            # Narrow UCS-2 build
+            pattern = re.compile(u'('
+                u'\ud83c[\udf00-\udfff]|'
+                u'\ud83d[\udc00-\ude4f\ude80-\udeff]|'
+                u'[\u2600-\u26FF\u2700-\u27BF])+', 
+                re.UNICODE)
+    else:
+    # Python 3+ should handle all strings as unicode. Right?
+        try:
+            # Wide UCS-4 build
+            pattern = re.compile('['
+                '\U0001F300-\U0001F64F'
+                '\U0001F680-\U0001F6FF'
+                '\u2600-\u26FF\u2700-\u27BF]+', 
+                re.UNICODE)
+        except re.error:
+            # Narrow UCS-2 build
+            pattern = re.compile('('
+                '\ud83c[\udf00-\udfff]|'
+                '\ud83d[\udc00-\ude4f\ude80-\udeff]|'
+                '[\u2600-\u26FF\u2700-\u27BF])+', 
+                re.UNICODE)
+
+    def replace(match):
```

Comment:
@BrendanAdkins [`emojize()`](https://github.com/carpedm20/emoji/blob/master/emoji/core.py#L41-L45) also has optional support for a set of aliases so it would be nice to see that here too.  The docstring for `demojize()` should also include a note about how multiple aliases reference the same unicode code, so when decoding they're not guaranteed to get the intended emoji name.

For example:

```python
print(emoji.emojize("This looks good :+1:!", use_aliases=True))
'This looks good \U0001f44d!'
```

But the unicode code `\U0001f44d` is aliased to the the official `:thumbs_up_sign:`, and `:+1` and `:thumbsup:`.  If we pass the same string through `emoji.demojize()` we're not guaranteed to get `:+1:` in the output string.  Instead we're guaranteed to get one of the three but it will be whichever one is added to the [`emoji.unicode_codes.UNICODE_EMOJI_ALIAS`](https://github.com/carpedm20/emoji/blob/master/emoji/unicode_codes.py#L2605) dictionary last.